pkg/utils/apply/apply.go

```diff
@@ -315,3 +330,43 @@ func DisableUpdateAnnotation() ApplyOption {
 		return nil
 	}
 }
+
+// SharedByApp let the resource be sharable
+func SharedByApp(app *v1beta1.Application) ApplyOption {
+	return func(act *applyAction, existing, desired client.Object) error {
+		// calculate the shared-by annotation
+		// if resource exists, add the current application into the resource shared-by field
+		var sharedBy string
+		if existing != nil && existing.GetAnnotations() != nil {
+			sharedBy = existing.GetAnnotations()[oam.AnnotationAppSharedBy]
+		}
+		sharedBy = AddSharer(sharedBy, app)
+		util.AddAnnotations(desired, map[string]string{oam.AnnotationAppSharedBy: sharedBy})
```

Comment:
will the annotation updated if the apply skipped?

Reply:
Yes, it will.